pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -1875,18 +1874,26 @@ func (s *AWSCloud) listPublicSubnetIDsinVPC(vpcId string) ([]string, error) {
 		return nil, err
 	}
 
+	var subnetIds []string
 	availabilityZones := sets.NewString()
 	for _, subnet := range subnets {
 		az := orEmpty(subnet.AvailabilityZone)
 		id := orEmpty(subnet.SubnetId)
-		if !isSubnetPublic(rt, id) {
-			glog.V(2).Infof("Ignoring private subnet %q", id)
-			continue
-		}
+
 		if availabilityZones.Has(az) {
 			glog.Warning("Found multiple subnets per AZ '", az, "', ignoring subnet '", id, "'")
 			continue
 		}
+
+		isPublic, err := isSubnetPublic(rt, id)
```

Comment:
Not sure why you moved this one?  Shouldn't we skip over non-public subnets before we warn about having to make an arbitrary choice between subnets?